app/upgrades/v20/upgrades.go

```diff
@@ -345,19 +396,18 @@ func MigrateConsumerAdditionProposal(
 }
 
 // metadataFromCAP returns ConsumerMetadata from a ConsumerAdditionProposal
-func metadataFromCAP(
-	prop *providertypes.ConsumerAdditionProposal,
-) (providertypes.ConsumerMetadata, error) {
+func metadataFromCAP(prop *providertypes.ConsumerAdditionProposal) providertypes.ConsumerMetadata {
 	metadata := providertypes.ConsumerMetadata{
 		Name:        prop.Title,
 		Description: prop.Description,
 		Metadata:    "TBA",
 	}
 	err := providertypes.ValidateConsumerMetadata(metadata)
-	// TODO: avoid returning an error here;
-	// for this, we need to make sure the validation works always
-	// (e.g., truncate the fields)
-	return metadata, err
+	if err != nil {
+		metadata.Name = providertypes.TruncateString(metadata.Name, providertypes.MaxNameLength)
+		metadata.Description = providertypes.TruncateString(metadata.Description, providertypes.MaxDescriptionLength)
```

Comment:
In case of error, do we still want `Metadata` to be set to `"TBA"`?

Reply:
Yeah, why not? We'll need to do this in a subsequent upgrade as we don't know what consumers will send props in this period.